WCFDataService/Client/System/Data/Services/Client/DataServiceContext.cs

```diff
@@ -77,6 +77,34 @@ public enum UndeclaredPropertyBehavior
     }
 
     /// <summary>
+    /// Indicates how DataServiceContext will check for uninitialized properties on the proxy types.
+    /// Proxies with lazy initialized properties (initialized in the property getter) can specify the naming convention for the backing fields so the DataServiceContext does not trigger initialization when not desired.
+    /// If no backing fields can be found, the property getters are used falling back to the "old" behaviour.
+    /// </summary>
+    public enum ProxyBackingFieldNamingConvention
+    {
+        /// <summary>
+        /// The default value, the DataServiceContext will first try ProxyBackingFieldNamingConvention.Underscores then ProxyBackingFieldNamingConvention.CamelCasing.
+        /// </summary>
+        Auto = 0,
+
+        /// <summary>
+        /// The DataServiceContext will try to find the backing field by assuming it is named like the property but prefixed with two or one underscores. This is the naming convention used in proxies generated with DataSvcUtil.exe
+        /// </summary>
+        Underscores = 1,
```

Comment:
A proxy generated with DataSvcUtil (or Connected Service) will have the fields eagerly initialized, like:
_Files = new DSC<File>(...);
Where _Files is a FIELD not a property.
So yes, if you want to make a standard auto generated proxy into one using lazy initializing, you need to do something, like shown with the regex.
If you on the other hand are writing your proxy by hand you can just write the lazy initializing your self. Two naming conventions are supported for detecting backing fields - camel casing and underscore prefixing.